Document how to transition between window and session animation loops.

toji opened the pull request:
/fixes #352

Due to the variations between how different platforms handle window
animation frames when immersive content is running, making sure that
apps handle the transition between the window and session animation
loops gracefully is unfortunately unintuitive. This adds an example that
demonstrates one way to do it correctly, while also explaining the
problem.

@NellWaliczek: I'm a bit torn on where to put this. It feels out-of-place
in the spec. I can easily move the text over to the explainer if you feel
it would be more appropriate, but I worry that it would be overlooked
there. Thoughts?

Manishearth commented on index.bs:
```diff
@@ -724,6 +725,61 @@ When an {{XRSession}} |session| receives updated [=viewer=] state from the [=/XR
 </div>
 
+<div class="example">
+Depending on the device being used, callbacks supplied to {{Window}} {{Window/requestAnimationFrame()}} may not be processed while and [=immersive session=] is active. (For instance, on a mobile or standalone device where the immersive content completely obscures the HTML document.) As such, developers should not rely on {{Window}} {{Window/requestAnimationFrame()}} callbacks to schedule {{XRSess ...
```
Some browsers support "warnings" in devtools. Should we have a non-normative note suggesting browsers show a warning for `session.rAF()` called within `window.rAF()` or vice versa?

toji replied:
I've wondered about it, but I'm not fully convinced that there's no legitimate use cases for something like that? I don't know, would have to give it some more thought.

NellWaliczek replied:
Heck, could we do something more drastic and reject `session.rAF()` if invoked from `window.rAF()`?  If that's too much, maybe we reject it from there if it's never been hooked before?

Thinking about the consequences of such a choice... in the future, some layers won't ever require processing a rAF (for example a hardware-backed video projection layer).... so what would happen when the video ends and the developer wants to start rendering their own content with a traditional layer type... when would we expect them to hook the `session.rAF()`?  I do worry they might try to use the `window.rAF()` if not prevented from doing so...

Manishearth replied:
> Heck, could we do something more drastic and reject session.rAF() if invoked from window.rAF()? 

I considered suggesting this but it felt a bit too drastic to me. I'm not against it though.

toji commented on index.bs:
```diff
@@ -724,6 +725,61 @@ When an {{XRSession}} |session| receives updated [=viewer=] state from the [=/XR
 </div>
 
+<div class="example">
+Depending on the device being used, callbacks supplied to {{Window}} {{Window/requestAnimationFrame()}} may not be processed while and [=immersive session=] is active. (For instance, on a mobile or standalone device where the immersive content completely obscures the HTML document.) As such, developers should not rely on {{Window}} {{Window/requestAnimationFrame()}} callbacks to schedule {{XRSess ...
```
We MAY say "must not" but we MUST NOT say "MUST NOT". 😉

toji commented:
> I'm still a bit confused by the sample code because I'm not sure the purpose of the windowAnimationSuspended variable.

It was to flag that a new `window.rAF` call needed to be scheduled when the session ends, but I agree that it was confusing. I refactored the code a bit to remove the need for that variable at the expense of potentially letting `window.rAF` spin without rendering during the session. Makes the code slightly more concise and focuses it on the core problem more.

Addressed the rest of your feedback as well. Any further thoughts?

toji commented:
Filed #736. Thanks!
